src/mon/FSCommands.cc

```diff
@@ -385,11 +385,6 @@ class SetHandler : public FileSystemCommandHandler
         {
           fs->mds_map.set_inline_data_enabled(true);
         });
-
-        // Update `compat`
-        CompatSet c = fsmap.get_compat();
-        c.incompat.insert(MDS_FEATURE_INCOMPAT_INLINE);
-        fsmap.update_compat(c);
```

Comment:
It's maintained by the `inline_data_enabled` setting on the `MDSMap`. We can't really know if a file system "ever" had it turned on. That's not really saved anywhere.

As it stands with this change, all file systems will add that incompat feature because all MDS (Pacific onwards) will advertise it.